### PR TITLE
[QT-309] Ensure environment variables are populated before proceeding

### PR DESCRIPTION
**Pull request by manthonygfp**
This PR represents the first body of work in the process of ensuring that credentials are set in order for certain acceptance tests to be run. There are some subtle distinctions between environment variables being set and the credentials being valid. This work only adds the former.

You will find a test helper that has been added in order to DRY up this process. The test helper is then used in acceptance tests that need cloud credentials set in their environment.

Some tests that need credentials are either not available in the Vault OSS repository or require extended logic to fully check that their credentials are not available. Those are not included in this PR though they are a part of the larger QT-309 task. I have provided a simple table that indicates what should be in this PR and what shouldn't.

image

"No failures" indicates that the only requirement for the test to not fail is that Docker must be locally available. We have decided that this is an implicit requirement.

### CODE REVIEW

**Review by joshbrand — APPROVED** [review]

File: helper/testhelpers/testhelpers.go [hunk 1]
```diff
@@ -974,3 +976,13 @@ func SetupLoginMFATOTP(t testing.T, client *api.Client) (*api.Client, string, st
 	SetupMFALoginEnforcement(t, client, enforcementConfig)
 	return entityClient, entityID, methodID
 }
+
+func SkipUnlessEnvVarsSet(t testing.T, envVars []string) {
+	t.Helper()
+
+	for _, i := range envVars {
+		if os.Getenv(i) == "" {
```
Comment (mikebaum) [inline helper/testhelpers/testhelpers.go:984]: Perhaps you should rather use:  `os.LookupEnv()`, which returns a string value and a bool that indicates if the env var was set, i.e.

```
  if _, isSet := os.LookupEnv(i); !isSet {
    ....
  }
```

File: plugins/database/redshift/redshift_test.go [hunk 2]
```diff
@@ -70,6 +71,14 @@ func TestRedshift_Initialize(t *testing.T) {
 		t.SkipNow()
 	}
 
+	// Ensure each cred is populated.
+	credNames := []string{
```
Comment (mikebaum) [inline plugins/database/redshift/redshift_test.go:75]: Why not create another var with this slice and use it throughout the file.

**Review by mikebaum — COMMENTED** [review]
Just a couple of minor suggestions, but overall this looks great.

File: helper/testhelpers/testhelpers.go (hunk 1 above)
Comment (manthonygfp) [reply helper/testhelpers/testhelpers.go:984]: This was a consideration, but the reason that I didn't go with it is that it doesn't provide me useful choices beyond complicating handling the return with no change to the outcome I was seeking. How does that sit with you?

**Review by joshbrand — APPROVED** [review]

**Pull request merged**